site-content/source/modules/ROOT/pages/development/dependencies.adoc

```diff
@@ -2,50 +2,83 @@
 
 == Dependency Management
 
-Managing libraries for Cassandra is a bit less straight forward compared
-to other projects, as the build process is based on ant, maven and
-manually managed jars. Make sure to follow the steps below carefully and
-pay attention to any emerging issues in the `ci` and reported related
-issues on Jira/ML, in case of any project dependency changes.
+New dependencies should not be included without community consensus first being
+obtained via a `[DISCUSS]` thread on the dev@cassandra.apache.org mailing list.
 
 As Cassandra is an Apache product, all included libraries must follow
 Apache's https://www.apache.org/legal/resolved.html[software license
 requirements].
 
-=== Required steps to add or update libraries
-
-* Add or replace jar file in `lib` directory
-* Add or update `lib/license` files
-* Update dependencies in `build.xml`
-** Add to `parent-pom` with correct version
-** Add to `all-pom` if simple Cassandra dependency (see below)
-
-=== POM file types
-
-* *parent-pom* - contains all dependencies with the respective version.
-All other poms will refer to the artifacts with specified versions
-listed here.
-* *build-deps-pom(-sources)* + *coverage-deps-pom* - used by `ant build`
-compile target. Listed dependenices will be resolved and copied to
-`build/lib/{jar,sources}` by executing the
-`maven-ant-tasks-retrieve-build` target. This should contain libraries
-that are required for build tools (grammar, docs, instrumentation), but
-are not shipped as part of the Cassandra distribution.
-* *test-deps-pom* - refered by `maven-ant-tasks-retrieve-test` to
-retrieve and save dependencies to `build/test/lib`. Exclusively used
-during JUnit test execution.
-* *all-pom* - pom for
-https://mvnrepository.com/artifact/org.apache.cassandra/cassandra-all[cassandra-all.jar]
-that can be installed or deployed to public maven repos via
-`ant publish`
+Cassandra uses the Ant build system and Maven POMs for dependency specification.
+In Cassandra 4.2 the format of POMs was moved from within the `build.xml` file to separate POM template files that are processed by Ant.
+In both pre-4.2 and post-4.2 Cassandra, there are several POMs that dependencies can be included in:
+
+* *parent-pom*
+    - Contains all dependencies with the respective version. All other poms
+      will refer to the artifacts with specified versions listed here.
+    - Since Cassandra 4.2, the `parent-pom` template is `.build/parent-pom-template.xml`.
+* *build-deps-pom(-sources)* + *coverage-deps-pom*
+    - used by the `ant build` target. Listed dependencies will be resolved and
+      copied to `build/lib/{jar,sources}` by executing the
+      `maven-ant-tasks-retrieve-build` target. This should contain libraries that are
+      required for build tools (grammar, docs, instrumentation), but are not
+      shipped as part of the Cassandra distribution.
+    - Since Cassandra 4.0, `coverage-deps-pom` has been removed and the
```

Comment:
add an item about  `lib/` and  `resolver-dist-lib` ?

Reply:
Can you share more about what you mean? I'll admit I haven't used resolver-dist-lib directly so I'm not the right person to document its behavior for new contributors.

Reply:
When you `ant artifacts` this will depend on the `resolver-dist-lib` target.  This propagates the `lib/` folder.  This folder contains the jar files that are actually distributed in a release (e.g. the tarball, the deb and rpm packages).  As opposed to `build/lib/jars` which contains more, e.g. build libs, provided scope, optional deps, etc.

We have to keep a closer eye on what gets generated into the `lib/` folder, according to ASF distribution policy: https://www.apache.org/legal/resolved.html

Reply:
I've added more details here